app/controllers/document_topics_controller.rb

```diff
@@ -12,5 +12,18 @@ class DocumentTopicsController < ApplicationController
   def edit
     @document = Document.find_by_param(params[:document_id])
     @topic_index = TopicsService.new.topic_index
+    @topic_content_ids, @version = TopicsService.new.topics_for_document(@document)
+  end
+
+  def update
+    document = Document.find_by_param(params[:document_id])
+    TopicsService.new.patch_topics(document, params.fetch(:topics, []), params[:version])
+    redirect_to document_path(document), notice: t("documents.show.flashes.topics_updated")
```

Comment:
I'm not sure of the answer but the naming structure we have for the locale key feels a bit weird with it being documents.show but that's entirely controlled by the documents topics controller.

Reply:
So the convention is that copy should be put in the locale file for the page it is shown on. Otherwise it's difficult for a content designer to work out which file a particular piece of copy might be in e.g. if I see an alert on the document summary page, the copy should be in documents/show.yml.